### ISSUE TITLE
[local_auth] Support native authentication on desktops

### ISSUE THREAD

**Issue by TheOneWithTheBraid** [flutter/flutter#70270]
## Use case

The `local_auth` package is useful when handling sensitive data in Flutter applications. In my case, I am processing encrypted, sensitive messages which I would obviously like to protect. Especially on Desktop systems in the professional environment, lots of sensitive data is being processed. That is why I would like to propose to support native authentication on desktop systems too.

## Proposal

Especially on Linux, the implementation should be done using `pam`, the native Linux authentication manager, as pam itself supports various authentication types, security levels. Especially, password authentication, facial detection and fingerprint detection can be natively integrated into pam.

About Windows and macOS, I am unfortunately not in knowledge to propose a suiting method.

**Comment by azchohfi** [comment]
I'm implementing this for Windows using [Windows::Security::Credentials::UI::UserConsentVerifier](https://docs.microsoft.com/en-us/uwp/api/windows.security.credentials.ui.userconsentverifier). The only problem is that this API does not let you check which biometric type is available, nor which one should be used, since the user can use *any* system registered/available method, depending on what is available at the time. Still, it is definitely the right Windows API.

**Comment by Jethro87** [comment]
It would be great to have macOS support. Any updates on if this is in the works?

**Comment by stuartmorgan** [comment]
We have moved to tracking new plugin implementations per-platform; since Windows is done and https://github.com/flutter/flutter/issues/140685 was filed for macOS, I'm going to adjust this to just be about Linux.

**Comment by suxoikorm** [comment]
@stuartmorgan Hello. Can I ask, you have removed the macos label, but the package has not yet been updated. release soon?

**Comment by Vivekananda-Mullangi** [comment]
@stuartmorgan is there any planned schedule for release of linux support?

**Comment by stuartmorgan** [comment]
No; please see https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#when-will-my-bug-be-fixed

**Comment by Vivekananda-Mullangi** [comment]
Sorry, I didnt know that. Won't happen again

**Comment by Skyost** [comment]
Hello,

I've just implemented local auth on Linux for my app. Feel free to use the code in your projects !

* Linux-side code : https://github.com/Skyost/OpenAuthenticator/blob/main/linux/my_application.cc
* Dart-side code : https://github.com/Skyost/OpenAuthenticator/blob/main/lib/utils/local_authentication/method_channel.dart

**Comment by prateekmedia** [comment]
@Skyost Could you please create a PR as well for this on [flutter/packages](https://github.com/flutter/packages), since you already have a working implementation with Polkit.

**Comment by stuartmorgan-g** [comment]
Given that there is not a single clear path forward here, the next step would be [a design document](https://github.com/flutter/flutter/blob/master/docs/contributing/Design-Documents.md), rather than a PR. Before any code could be reviewed for flutter/packages there would need to be agreement among key stakeholders about what foundation to build the implementation on, and a design doc that lays out the options and the pros and cons of each would be the place to do that.
